pkg/server/registry/apigroups/acorn/apps/validator.go

```diff
@@ -584,13 +584,22 @@ func (s *Validator) checkScheduling(ctx context.Context, params *apiv1.App, proj
 		return append(validationErrors, field.Invalid(field.NewPath("spec", "image"), params.Spec.Image, fmt.Sprintf("error listing compute classes: %v", err)))
 	}
 
+	// filter compute classes that don't match the relevant regions
+	filteredComputeClasses := new(apiv1.ComputeClassList)
+	for _, cc := range computeClasses.Items {
+		// if the region is set in the spec then we aren't concerned with the defaultRegion
+		if (params.Spec.Region != "" && slices.Contains(cc.SupportedRegions, params.Spec.Region)) || slices.Contains(cc.SupportedRegions, defaultRegion) {
```

Comment:
That's a good point. The logic here should be to only do this check if the `cc.SupportedRegions` is non-empty.

Reply:
I think checking whether `cc.SupportedRegions` is empty is fine, given we know that it is only set in manager api, and then we should do the filtered logic here to make sure region matches. Otherwise we will just use the existing computeclasses.

Will leave @thedadams for some opinions.

Reply:
There is a [controller](https://github.com/acorn-io/runtime/blob/0be16dddfaa3727c3407b066d75232d24d6e508e/pkg/project/handlers.go#L17) in runtime that will ensure that projects have their default region set. Also, we we [sync](https://github.com/acorn-io/manager/blob/9dfc5e69e81aae2e1242b4906219820b72137d00/modules/cluster-agent/cluster-account-agent/pkg/controller/project/handlers.go#L39-L43) projects downstream, we also set the default region.

So I think this is good.